src/main/java/ch/njol/skript/expressions/ExprBeaconValues.java

```diff
@@ -0,0 +1,219 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Getter;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.Beacon;
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffectType;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Beacon Effects")
+@Description({
+	"The active effects of the beacon.",
+	"The secondary effect can only be set to regeneration and it can only be set when the beacon has achieved the highest tier.",
+	"You can only change the range and clear the primary and secondary effects on Paper."
+})
+@Examples({
+	"broadcast tier of {_block}",
+	"set primary effect of {_block} to haste",
+	"add 1 to range of {_block}"
+})
+@Since("INSERT VERSION")
+public class ExprBeaconValues extends PropertyExpression<Block, Object> {
+
+	private static boolean PAPER_METHOD = false;
+
+	static {
+		if (Skript.methodExists(Beacon.class, "getTier") && Skript.methodExists(Beacon.class, "setPrimaryEffect", PotionEffectType.class)) {
+			PAPER_METHOD = true;
+		}
+
+		Skript.registerExpression(ExprBeaconValues.class, Object.class, ExpressionType.PROPERTY,
+			"%blocks%['s] primary effect",
+			"primary effect of %blocks%",
+			"%blocks%['s] secondary effect",
+			"secondary effect of %blocks%",
+			"%blocks%['s] range",
+			"range of %blocks%",
+			"%blocks%['s] tier",
+			"tier of %blocks%"
```

Comment:
You might prefer to use `(primary:...|secondary:...|...)` parse tags here rather than fiddling about with the matched pattern later.